trigger/patch.go

```diff
@@ -118,45 +118,63 @@ func (t *patchTriggers) patchOutcome(sub *event.Subscription) (*notification.Not
 		}
 	}
 
-	if sub.Subscriber.Type == event.GithubPullRequestSubscriberType {
-		target, ok := sub.Subscriber.Target.(*event.GithubPullRequestSubscriber)
-		if !ok {
-			return nil, errors.Errorf("target '%s' didn't not have expected type", sub.Subscriber.Target)
-		}
-		subType := target.Type
+	isReady, err := t.waitOnChildrenOrSiblings(sub)
+	if err != nil {
+		return nil, err
+	}
+	if !isReady {
+		return nil, nil
+	}
 
-		if t.patch.IsParent() || (t.patch.IsChild() && subType == event.WaitOnChild) {
-			// get the children or siblings to wait on
-			childrenOrSiblings, parentPatch, err := t.patch.GetPatchFamily()
-			if err != nil {
-				return nil, errors.Wrap(err, "error getting child or sibling patches")
-			}
+	return t.generate(sub)
+}
 
-			// make sure the parent is done, if not, wait for the parent
-			if t.patch.IsChild() {
-				if !evergreen.IsFinishedPatchStatus(parentPatch.Status) {
-					return nil, nil
-				}
-			}
-			childrenStatus, err := getChildrenOrSiblingsReadiness(childrenOrSiblings)
-			if err != nil {
-				return nil, errors.Wrap(err, "error getting child or sibling information")
-			}
-			if !evergreen.IsFinishedPatchStatus(childrenStatus) {
-				return nil, nil
-			}
+func (t *patchTriggers) waitOnChildrenOrSiblings(sub *event.Subscription) (bool, error) {
+	if sub.Subscriber.Type != event.GithubPullRequestSubscriberType {
+		return true, nil
+	}
+	target, ok := sub.Subscriber.Target.(*event.GithubPullRequestSubscriber)
+	if !ok {
+		return false, errors.Errorf("target '%s' didn't not have expected type", sub.Subscriber.Target)
+	}
+	subType := target.Type
 
-			if childrenStatus == evergreen.PatchFailed {
-				t.data.Status = evergreen.PatchFailed
-			}
+	if !(t.patch.IsParent() || (t.patch.IsChild() && subType == event.WaitOnChild)) {
```

Comment:
if the patch isn't a parent  and the patch isn't a child OR it's not subscription type wait on child? Does this imply "if it's a normal patch, return true?" Can we drop a comment to explain this boolean logic?

Reply:
(t.patch.IsChild() && subType == event.WaitOnChild) is in parentheses  so either: 

- it's a parent 
- its a child and it's also event.WaitOnChild

It's set up this way because the children, on outcome, send their status to the pr so it can show up in the pr as they are running (similar to builds). That's why, if it's a child, we only want to wait on its siblings if it's event.WaitOnChild. 

I'll add a comment.